src/Widgets/AppContainers/AbstractAppContainer.vala

```diff
@@ -108,7 +108,34 @@ namespace AppCenter {
             image = new Gtk.Image ();
 
             action_button = new Widgets.HumbleButton ();
-            action_button.download_requested.connect (() => action_clicked.begin ());
+            
+            var project_license = package.component.project_license;
+            bool license_proprietary = false;
+            string? license_copy;
+            string? license_url;
+
+            if (project_license != null && project_license.has_prefix ("LicenseRef")) {
+                license_proprietary = true;
+                
+                // i.e. `LicenseRef-proprietary=https://example.com`
+                var split_license = project_license.split_set ("=", 2);
+
+                license_url = split_license[1];
```

Comment:
This isn't guaranteed to be anything, because there might not be an `=foo` in `project_license`.